module/data/fields.mjs

```diff
@@ -47,28 +47,36 @@ export class AdvancementDataField extends foundry.data.fields.ObjectField {
 /**
  * @typedef {StringFieldOptions} FormulaFieldOptions
  * @property {boolean} [deterministic=false]  Is this formula not allowed to have dice values?
+ * @property {boolean} [extended=false]       Does this formula support conditional operators (aka max uses field)?
  */
 
 /**
  * Special case StringField which represents a formula.
  *
  * @param {FormulaFieldOptions} [options={}]  Options which configure the behavior of the field.
  * @property {boolean} deterministic=false    Is this formula not allowed to have dice values?
+ * @property {boolean} extended=false         Does this formula support conditional operators (aka max uses field)?
```

Comment:
Is it not the case that an `extended` formula must necessarily be `deterministic` too? Would it make sense to evaluate all deterministic `FormulaField`s with `safeEval` and save us the extra option?

Reply:
Well, at the moment we have formula fields (like passive skill bonus) that are deterministic but don't support the extended math properties (so we have non-deterministic roll formulas, deterministic roll formulas, and deterministic math formulas).

That said, we are currently relying on `Roll#evaluate({async: false})` for those bonuses, so it might be worth revisiting the `simplifyBonus` code. Not sure if we want to change it to `SafeEval`, which would allow for those extended math operations, change `simplifyBonus` to `async`, or wait for core to provide a synchronous evaluation method for deterministic rolls.

Reply:
The synchronous evaluation method for deterministic rolls *is* `safeEval`. I think it's probably an unintentional side-effect that it allows for ternary expressions, but so long as it's at least a strict superset of what can be evaluated by `Roll#evaluate`, it's probably safe to use it.

Would it make sense to use `safeEval` here to simplify the API surface and just make a note that it will technically validate certain things correctly even though they might fail later in execution? We can iron out the issues around synchronous evaluation at a later date when we decide what to do in core. It shouldn't be a functionality regression since we don't validate anything currently anyway.

Reply:
Okay, one more pass. Now all "deterministic" `FormulaField`s go through `safeEval`. I left in the explicit `isDeterministic` check for a clearer error message.